### PR TITLE
Implemented new flow

### PR DESCRIPTION
**Pull request by andrii-onufriichuk**

### CODE REVIEW

File: Api/UpdateInvoiceInterface.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+interface UpdateInvoiceInterface
+{
+    /**
+     * @param string $invoiceId
+     * @param string $accountId
+     * @param string $companyId
+     * @return void
+     */
+    public function updateInvoiceById(string $invoiceId, string $accountId, string $companyId): void;
```
Comment (jay-rvvup) [inline Api/UpdateInvoiceInterface.php]: Please can this be changed to take in a body with this structure
```
{
  "statementId": "string",
  "invoiceId": "string",
  "accountId": "string",
  "companyId": "string"
  "paymentDate": "string",
  "paymentMethod": "string",
  "paymentReference": "string",
  "total": {
    "amount": "string",
    "currency": "string"
  },
  "invoices": [
    {
      "reference": "string",
      "amountPaid": {
        "amount": "string",
        "currency": "string"
      }
    }
  ]
}
```

File: Block/Adminhtml/Field/DynamicRows.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+use Magento\Config\Block\System\Config\Form\Field\FieldArray\AbstractFieldArray;
+use Magento\Framework\DataObject;
+
+class DynamicRows extends AbstractFieldArray
+{
+    protected function _prepareToRender()
+    {
+        $this->addColumn('company', ['label' => __('Company'), 'class' => 'required-entry']);
+        $this->addColumn('jwt_key', ['label' => __('JWT Key'), 'class' => 'required-entry']);
```
Comment (jay-rvvup) [inline Block/Adminhtml/Field/DynamicRows.php]: Please can this be updated to
```
        $this->addColumn('api_key', ['label' => __('API Key'), 'class' => 'required-entry']);
```

File: Api/InvoiceInterface.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+declare(strict_types=1);
+
+namespace Rvvup\AxInvoicePayment\Api;
+
+use Rvvup\AxInvoicePayment\Api\InvoiceDataInterface;
+
+interface InvoiceInterface
+{
+    const STATEMENT_ID = 'statementId';
+    const ACCOUNT_ID = 'accountId';
```
Comment (jay-rvvup) [inline Api/InvoiceInterface.php:11]: Please could we add companyId to this invoice object too

File: Api/UpdateInvoiceInterface.php (hunk 1 above)
Comment (andrii-onufriichuk) [reply Api/UpdateInvoiceInterface.php]: Added one parameter named 'data' of string type

Comment (jay-rvvup) [reply Api/UpdateInvoiceInterface.php]: But if its just a generic data string then how will the implementor of the interface know what data to expect? 
Thats why I was suggesting a more strongly typed input. 

But we can also just proceed like this and change it later if needed.

File: Controller/Process/Pay.php [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+     */
+    public function execute()
+    {
+        $statementId = $this->request->getParam('statement_id');
+        $companyId = $this->request->getParam('company_id');
+        $accountNumber = $this->request->getParam('account_number');
+        $result = $this->resultFactory->create(ResultFactory::TYPE_JSON);
+
+        try {
+            $url = $this->createCheckout(
```
Comment (jay-rvvup) [inline Controller/Process/Pay.php]: This could be "createAccountStatement"

**Review by jay-rvvup — APPROVED** [review]

**Pull request merged**